configure.ac

```diff
@@ -110,6 +110,16 @@ AS_IF([test "x$enable_gcc_checking" != xno],
 	[AC_SUBST(gcc_checking, --enable-checking=yes)],
 	[AC_SUBST(gcc_checking, --enable-checking=release)])
 
+AC_ARG_WITH(cmodel,
+	[AS_HELP_STRING([--with-cmodel],
+		[Select the code model to use when building libc and libgcc @<:@--with-cmodel=medlow@:>@])],
+	[],
+	[enable_gcc_checking=yes]
+	)
+AS_IF([test "x$with_cmodel" != x],
+	[AC_SUBST(cmodel, -mcmodel=$with_cmodel)],
+	[AC_SUBST(cmodel, -mcmodel=medlow)])
```

Comment:
This ends up getting passed to GCC which has helpful error messages now (it'll give you the legal options, for example).  It means the error is a bit later, but we don't have to duplicate the help text.  This is consistent with the other autoconf arguments.